ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Impl.hs

```diff
@@ -518,8 +522,8 @@ getEpochSlot hasFS hashInfo err curEpochInfo getWhat epochSlot =
           GetHeader -> fmap (headerHash,) $
               -- Get just the header
               withFile hasFS epochFile ReadMode $ \eHnd ->
-                -- TODO we cannot check the checksum in this case, as we're
-                -- not reading the whole block
+                -- We cannot check the checksum in this case, as we're not
+                -- reading the whole block
                 hGetExactlyAt hasFS eHnd size offset
```

Comment:
Introduce `hGetExactlyAtCRC` (?). That will do basically the same thing as we do now: read everything first and then compute the checksum. So we can later optimise this (or change it to allow incremental parsing) in one place.